shellwhat_ext/__init__.py

```diff
@@ -3,8 +3,38 @@
 from getopt import getopt, GetoptError
 from protowhat.Test import TestFail
 from shellwhat.sct_syntax import state_dec
+from shellwhat.checks import test_expr_error
 
-__version__ = '0.2.0'
+__version__ = '0.3.0'
+
+#-------------------------------------------------------------------------------
+
+@state_dec
+def test_condition(state, condition, msg):
+    '''Check whether a Boolean condition is satisfied, and if not,
+    report the error.  This can be used for tests like:
+
+        test_condition('.filiprc' in os.listdir('/home/repl'),
+                       "Home directory does not contain a .filiprc file")
+    '''
+
+    if not condition:
+        state.do_test(msg)
+    return state # all good
+
+#-------------------------------------------------------------------------------
+
+@state_dec
+def test_cwd(state, expected, msg=None):
+    '''Check whether the user is currently in the expected directory.
+    This is a convenience wrapper around a rather obscure shell check:
+    see https://github.com/datacamp/learn-bugs/issues/62 for the reason
+    that the shell expression is needed.
+    '''
+
+    if msg is None:
+        msg = 'Not in expected directory "{}"'.format(expected)
+    test_expr_error(state, "[[ $PWD == '{}' ]]".format(expected), msg=msg)
```

Comment:
I'm not convinced about this one. Can you add some unit tests that would show it's used?